airflow/www/security_manager.py

```diff
@@ -200,7 +200,7 @@ def get_dag_id_from_task_instance(resource_pk):
                 return None
             composite_pk = json.loads(resource_pk)
             ti = session.scalar(
-                select(DagRun)
+                select(TaskInstance)
```

Comment:
Maybe we should just do `select(TaskInstance.dag_id)` here since we really onlt use the field below?

Reply:
Yeah, it could reduce the amount of the data send between DB and Airflow.